pkg/commands/entrypoints/entrypoints.go

```diff
@@ -0,0 +1,251 @@
+
+	if c.flagFormat == "" {
+		c.flagFormat = "text"
+	}
+
+	if _, ok := allowedFormats[c.flagFormat]; !ok {
+		return fmt.Errorf("invalid format flag: %s (supported formats are: %s)", c.flagFormat, strings.Join(maps.Keys(allowedFormats), ", "))
+	}
+
+	logger.DebugContext(ctx, "finding entrypoint directories")
+
+	entrypoints, err := terraform.GetEntrypointDirectories(c.directory)
+	if err != nil {
+		return fmt.Errorf("failed to find terraform directories: %w", err)
+	}
+
+	entrypointDirs := make([]string, 0, len(entrypoints))
+	for _, e := range entrypoints {
+		entrypointDirs = append(entrypointDirs, e.Path)
+	}
+
+	logger.DebugContext(ctx, "terraform entrypoint directories", "entrypoint_dirs", entrypoints)
+
+	if !c.flagSkipDetectChanges {
+		logger.DebugContext(ctx, "finding git diff directories")
+
+		diffDirs, err := c.gitClient.DiffDirsAbs(ctx, c.flagDestRef, c.flagSourceRef)
+		if err != nil {
+			return fmt.Errorf("failed to find git diff directories: %w", err)
+		}
+		logger.DebugContext(ctx, "git diff directories", "directories", diffDirs)
+
+		moduleUsageGraph, err := terraform.ModuleUsage(ctx, c.directory, !c.flagFailUnresolvableModules)
+		if err != nil {
+			return fmt.Errorf("failed to get module usage for %s: %w", c.directory, err)
+		}
+
+		modifiedEntrypoints := make(map[string]struct{})
+
+		for _, changedFile := range diffDirs {
+			if entrypoints, ok := moduleUsageGraph.ModulesToEntrypoints[changedFile]; ok {
+				for entrypoint := range entrypoints {
+					modifiedEntrypoints[entrypoint] = struct{}{}
+				}
+			}
+			if _, ok := moduleUsageGraph.EntrypointToModules[changedFile]; ok {
+				modifiedEntrypoints[changedFile] = struct{}{}
+			}
+		}
+
+		files := maps.Keys(modifiedEntrypoints)
+		sort.Strings(files)
+
+		entrypointDirs = files
+	}
+
+	logger.DebugContext(ctx, "target directories", "target_directories", entrypointDirs)
+
+	if err := c.writeOutput(entrypointDirs); err != nil {
+		return fmt.Errorf("failed to write output: %w", err)
+	}
+
+	return nil
+}
+
+// writeOutput writes the command output.
+func (c *EntrypointsCommand) writeOutput(dirs []string) error {
+	cwd, err := c.WorkingDir()
+	if err != nil {
+		return fmt.Errorf("failed to get current working directory: %w", err)
+	}
+
+	// convert to child path for output
+	// using absolute path creates an ugly github workflow name
+	for k, dir := range dirs {
+		childPath, err := util.ChildPath(cwd, dir)
+		if err != nil {
+			return fmt.Errorf("failed to get child path for: %w", err)
+		}
+		dirs[k] = childPath
+	}
+
+	switch v := strings.TrimSpace(strings.ToLower(c.flagFormat)); v {
+	case "json":
+		if err := json.NewEncoder(c.Stdout()).Encode(dirs); err != nil {
+			return fmt.Errorf("failed to create json string: %w", err)
+		}
+	case "text":
+		for _, dir := range dirs {
+			c.Outf("%s", dir)
+		}
+	default:
+		return fmt.Errorf("invalid format flag: %s (supported formats are: %s)", c.flagFormat, strings.Join(maps.Keys(allowedFormats), ", "))
```

Comment:
I think the string list should be moved to a private global var, given the frequency of use.